jdbc/src/main/java/org/openconnectors/source/TableQuerier.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.openconnectors.source;
+
+import org.openconnectors.data.Record;
+import org.openconnectors.data.TableMetaData;
+import org.openconnectors.util.ConnectionProvider;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class TableQuerier {
+    protected final String tableName;
+    protected TableMetaData tableMetaData;
+    protected String schemaPattern;
+    protected PreparedStatement preparedStatement;
+    private ConnectionProvider connectionProvider;
+
+    public TableQuerier(ConnectionProvider connectionProvider, String schemaPattern, String tableName) {
+        if (tableName == null || tableName.isEmpty()) {
+            throw new IllegalArgumentException("TableName cannot be null or empty.");
+        }
+        this.tableName = tableName;
+        this.schemaPattern = schemaPattern;
+        this.connectionProvider = connectionProvider;
+    }
+
+    public List<Record> query() throws SQLException {
+        ResultSet resultSet = getPreparedStatement().executeQuery();
+        int columnsCount = tableMetaData.getColumnsCount();
+        List<Record> records = new ArrayList<>();
+        Object[] values;
+        while (resultSet.next()) {
+            values = new Object[columnsCount];
+            for (int i = 1; i <= columnsCount; i++) {
+                values[i - 1] = resultSet.getObject(i);
+            }
+            records.add(new Record(values, tableMetaData));
+        }
+        return records;
+    }
+
+    public PreparedStatement getPreparedStatement() throws SQLException {
+        if (preparedStatement == null) {
+            preparedStatement = createPreparedStatement();
+            createTableMetaData();
```

Comment:
Would be good to allow per driver type specific configs.  
For Oracle `setFetchSize` is very important. It defaults to 10. might be enough on low volume update but it will be a performance bottleneck for initial load or larger volume updates.

Reply:
Could you please state the full list of configs you think are important?